NPPESAPI.net/NPPESApiClient.cs

```diff
@@ -17,7 +17,7 @@ public static class NPPESApiClient
     {
         private static readonly Lazy<HttpClient> defaultClient;
         private static readonly Lazy<JsonSerializer> serializer;
-        private const string BASE_ADDRESS_PATH = "https://npiregistry.cms.hhs.gov/api/resultsDemo2/";
+        private const string BASE_ADDRESS_PATH = "https://npiregistry.cms.hhs.gov/api/?version=2.1";
```

Comment:
After running tests it appears this may be a problem here:

If you look at [NPPESRequest](https://github.com/ForcuraCo/npi-api-net/blob/020825281f2a5d510af1163e5ce0fd38ba700d3a/NPPESAPI.net/NPPESRequest.cs#L111) you'll see we form our querystring here and append to the `BASE_ADDRESS_PATH`.  This results in the client making a request in the form:

`https://npiregistry.cms.hhs.gov/api/?version=2.1?number=1234567890&limit=10&skip=5`

Because of the invalid format we receive a generic error which is also in a different format than we typically expect (Looks like lack of standardization in the api there, the errors property is no longer an array, it is an object in this case).

I'm thinking it may be best to add `Version` as a property on the `NPPESRequest` which would default unless overridden using the builder.  This would then form a valid request url.

The downside would be, when a user upgrades their client, they would be forcefully placed on the latest NPPES api version, unless they specify which version in their application.

What are your thoughts here?

Reply:
I'm not sure how I overlooked that-probably because it was somehow working for me during my testing. Using the bad format, I get results back and no errors in our client app (Xamarin iOS), but making the same request in postman, I get the error. Very puzzling. I do, however, get back the error message in the expected format (array of error objects):
`{
    "Errors": [
        {
            "description": "Unsupported Version",
            "field": "version",
            "number": "17"
        }
    ]
}`

I agree it would be best to add a `Version` property to the `NPPESRequest`. I'm thinking of adding an enum with the different available versions to make it easy for users to specify which version they would like. I don't see a problem defaulting to the latest version, at least currently, as there are no breaking changes from 2.0 to 2.1 as far as I can tell.